src/frequenz/dispatch/_dispatcher.py

```diff
@@ -175,20 +175,26 @@ async def run():
     """
 
     def __init__(
-        self, microgrid_id: int, grpc_channel: grpc.aio.Channel, svc_addr: str
+        self,
+        *,
+        microgrid_id: int,
+        grpc_channel: grpc.aio.Channel,
+        svc_addr: str,
+        key: str,
     ):
         """Initialize the dispatcher.
 
         Args:
             microgrid_id: The microgrid id.
             grpc_channel: The gRPC channel.
             svc_addr: The service address.
+            key: The key to access the service.
         """
         self._running_state_channel = Broadcast[Dispatch](name="running_state_change")
         self._lifecycle_events_channel = Broadcast[DispatchEvent](
             name="lifecycle_events"
         )
-        self._client = Client(grpc_channel, svc_addr)
+        self._client = Client(grpc_channel=grpc_channel, svc_addr=svc_addr, key=key)
```

Comment:
Looks good to me.
Just side note that 3 out of 4 arguments to Dispatcher constructor are really arguments to Client constructor. 
And they are just passed to Client. And every description for these arguments is repeated in Client. 
Maybe we could have single struct like ClientArgs in the future?

Reply:
That could work, though we want to change to the [base-client](https://github.com/frequenz-floss/frequenz-client-base-python/blob/v0.x.x/src/frequenz/client/base/client.py#L116) where it will do this job for us